src/ngraph/onnx_import/mapping.hpp

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright 2017-2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#include <map>
+#include <ostream>
+#include "ngraph/type/element_type.hpp"
+#include "onnx.pb.h"
+
+namespace ngraph
+{
+    namespace onnx_import
+    {
+        struct mapping
+        {
+            static const std::map<onnx::TensorProto_DataType, ngraph::element::Type>
+                onnx_to_ng_types;
+        };
+
+        const std::map<onnx::TensorProto_DataType, ngraph::element::Type> mapping::onnx_to_ng_types{
```

Comment:
This mapping is overkill. Furthermore the static object creation may cause exception to be thrown which is uncatchable. One approach to solve the issue is to use `switch()` statement:
```
void to_tensorproto_datatype(onnx::TensorProto_DataType dt) {
    switch (dt) {
    case onnx::TensorProto_DataType_BOOL:
        return ngraph::element::boolean;
    ...
    }
}
```
The other approach would be to implement casting operator between two types. The advantage of casting operator is the result will be resolved in compile time for const expressions.

Reply:
Done using `switch` for now.